doc/code/qml_pulse.rst

```diff
@@ -0,0 +1,215 @@
+before parametrized terms to prevent discrepancy in the wire order.
+
+
+ParametrizedEvolution
+---------------------
+During a pulse sequence spanning time :math:`(t_0, t_1)`, the state evolves according to the time-dependent Schrodinger equation
+
+.. math:: \frac{\partial}{\partial t} |\psi\rangle = -i H(t) |\psi\rangle
+
+realizing a unitary evolution :math:`U(t_0, t_1)` of the input state, i.e.
+
+.. math:: |\psi(t_1)\rangle = U(t_0, t_1) |\psi(t_0)\rangle
+
+A :class:`~.ParametrizedEvolution` is this solution :math:`U(t_0, t_1)` to the time-dependent
+Schrodinger equation for a :class:`~.ParametrizedHamiltonian`.
+
+.. note::
+    The :class:`~.ParametrizedHamiltonian` must be Hermitian at all times. This is not explicitly
+    checked; ensuring a correctly defined Hamiltonian is the responsibility of the user.
+
+The :class:`~.ParametrizedEvolution` class uses a numerical ordinary differential equation
+solver (`here <https://github.com/google/jax/blob/main/jax/experimental/ode.py>`_). It
+can be created using the :func:`~.functions.evolve` function:
+
+.. code-block:: python
+
+    from jax import numpy as jnp
+
+    f1 = lambda p, t: p * jnp.sin(t) * (t - 1)
+    H = 2 * qml.PauliX(0) + f1 * qml.PauliY(1)
+    ev = qml.evolve(H)
+    ev
+    >>> ParametrizedEvolution(wires=[0, 1])
+
+The initial :class:`~.ParametrizedEvolution` does not have parameters defined, and so will
+not have a matrix defined. To obtain an :class:`~.Operator` with a matrix, it must be passed
+parameters and a time interval:
+
+.. code-block:: python
+
+    ev([1.2], t=[0, 4]).matrix()
+    >>> Array([[-0.14115842+0.j        ,  0.03528605+0.j        ,
+                 0.        -0.95982337j,  0.        +0.23993255j],
+               [-0.03528605+0.j        , -0.14115842+0.j        ,
+                 0.        -0.23993255j,  0.        -0.95982337j],
+               [ 0.        -0.95982337j,  0.        +0.23993255j,
+                -0.14115842+0.j        ,  0.03528605+0.j        ],
+               [ 0.        -0.23993255j,  0.        -0.95982337j,
+                -0.03528605+0.j        , -0.14115842+0.j        ]],      dtype=complex64)
+
+
+The parameters can be updated by calling the :class:`~.ParametrizedEvolution` again with different inputs.
+
+Additional options with regards to how the matrix is calculated can be passed to the :class:`.ParametrizedEvolution`
+along with the parameters, as keyword arguments:
+
+.. code-block:: python
+
+    qml.evolve(H)(params=[1.2], t=[0, 4], atol=1e-6, mxstep=1)
+
+These options are:
+
+- ``atol (float, optional)``: Absolute error tolerance
+- ``rtol (float, optional)``: Relative error tolerance
+- ``mxstep (int, optional)``: maximum number of steps to take for each time point
+- ``hmax (float, optional)``: maximum step size
+
+If not specified, they will default to predetermined values. See :class:`.ParametrizedEvolution` for details.
```

Comment:
@lillian542 it might make sense _not_ to list the function arguments here, and instead simply direct readers to the docstring. This will (a) reduce duplication and ensure there is one central source of truth, and (b) ensure that the description of arguments is closer to the code and more likely to be maintained.